### ISSUE TITLE
Connection reset by peer

### ISSUE THREAD

**Issue by yorickdewid** [xybu/onedrive-d-old#83]
Hello,
Just downloaded the last commit in dev (https://github.com/xybu/onedrive-d/commit/e050073c10e07085e1beb34a27e119015edb0721), and I noticed after downloading a low of data (2G) an exception appears in the worker thread.

``` python
Exception in thread worker-0:
Traceback (most recent call last):
  File "/usr/local/lib/python3.2/dist-packages/requests-2.4.3-py3.2.egg/requests/packages/urllib3/connectionpool.py", line 516, in urlopen
    body=body, headers=headers)
  File "/usr/local/lib/python3.2/dist-packages/requests-2.4.3-py3.2.egg/requests/packages/urllib3/connectionpool.py", line 308, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/lib/python3.2/http/client.py", line 970, in request
    self._send_request(method, url, body, headers)
  File "/usr/lib/python3.2/http/client.py", line 1008, in _send_request
    self.endheaders(body)
  File "/usr/lib/python3.2/http/client.py", line 966, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python3.2/http/client.py", line 815, in _send_output
    self.send(message_body)
  File "/usr/lib/python3.2/http/client.py", line 777, in send
    self.sock.sendall(datablock)
  File "/usr/lib/python3.2/ssl.py", line 371, in sendall
    v = self.send(data[count:])
  File "/usr/lib/python3.2/ssl.py", line 338, in send
    v = self._sslobj.write(data)
socket.error: [Errno 104] Connection reset by peer
```

Which causes multiple other exceptions.
After the socket error all worker threads fail and only the syncer continues.
##### Replay the error

Run the daemon and download quite a lot of data, after about 20 minutes the exception appears.

**Comment by xybu** [comment]
There are some limits on OneDrive server side just like #41 but they are not listed explicitly. If possible, could you estimate how many files the program fetched, and how large are those files, before the OneDrive server refuses to serve?

**Comment by yorickdewid** [comment]
I've reinstalled onedrive-d with a clean config. After exactly 6 and a half minute the first connection gets reset. Up to that point the daemon downloaded 130 directories and 53 files together 281Mb (mostly pictures). The exception occurs in the `get()` in live_api.py. I added a SocketError exception that catches the refused connection, so it can be handled in a gentle way. A possible solution would be to re-initiate the worker whenever a certain flag is set.
